criu/img-remote.c

```diff
@@ -950,7 +950,7 @@ int read_remote_image_connection(char *snapshot_id, char *path)
 		return -1;
 	}
 
-	if (!error || (snapshot_id[0] == NULL_SNAPSHOT_ID && path[0] != FINISH))
+	if (!error)
 		return sockfd;
```

Comment:
With this change, the following `if (error == ENOENT) ` becomes redundant?

Reply:
I don't think so. If error is ENOENT, we return -ENOENT, otherwise, we return -1 for all other errors.
We could return -error, but I don't recommend. It's better to control what kind of errors we allow to be returned.

Reply:
You are right. Thank you for clarifying.